test/ptp/tgm/functional/tests/tgm.go

```diff
@@ -64,7 +64,7 @@ var _ = Describe("PTP T-GM", func() {
 		It("Should check GNSS signal from PTP daemon log", func() {
 			_, err := pods.GetLog(client, ptpRunningPods[0], pkg.PtpContainerName)
 			Expect(err).NotTo(HaveOccurred(), "Error to find GNSS log in PTP daemon due to %s", err)
-			result := pods.WaitUntilLogIsDetectedRegex(client, ptpRunningPods[0], pkg.Timeout10Seconds, "nmea sentence: GNRMC(.*)")
+			result := pods.WaitUntilLogIsDetectedRegex(client, ptpRunningPods[0], pkg.TimeoutIn10Minutes, "nmea sentence: GNRMC(.*)")
```

Comment:
10 seconds to 10 minutes is a huge jump, and needs more explanation in the commit message. How was 10 minutes settled on?

This looks like a manifestation of something we've seen several times in SLCM as well and not yet found an adequate solution for a pass/fail test against an eventually consistent system. That issue is "How should you handle long durations of eventually?"
The temptation is always to increase the timeout to let the system get to the state the test is actually meant to test. However since the platform is only eventually consistent you can't guarantee it will **ever** get there. Not getting to that state quickly is not necessarily a failure of the *test* though. What should the general approach for the PTP tests be? @jnunyez might be able to say.

Would a ginkgo `Skip` if it takes more than e.g. 1 minute be more representative of this case? Meaning "system did not reach testable state fast enough, moving on"

Reply:
When extending the timeout I was thinking about having a more user friendly test, for a user who applies a config and running the tests just after. I was thinking about 10 min since it's a reasonable time (in my opinion) to wait for the config to be ready but not letting the user wait too much before getting the tests results (like an hour).

If you use the `Skip` method how do you know if the test failed since the config was not ready yet or because there was a different error so the config is not available and will never be?

Reply:
I communicated poorly, sorry.

The problem as I see it is that two, quite separate, things are being tested in this test: does the system get into the right state (config propagated, applied, pods restarted) in an acceptable time (10 minutes), and does GPS sync work when the system is in that state. As I understand it the intent is just to test if GPS sync works.

I think the config change should either be explicitly not part of the test suite (the test does not wait for configuration to propagate and it's documented that the cluster is tested as-is) or explicitly part (it gets its own test to test config is set).

I may not be understanding the separation of the layers in question here, in which case do leave it as is, but I think the choice to wait for so long for the system to get to the right state is a change in scope of the testing.

Reply:
> The temptation is always to increase the timeout to let the system get to the state the test is actually meant to test. However since the platform is only eventually consistent you can't guarantee it will **ever** get there. Not getting to that state quickly is not necessarily a failure of the _test_ though. What should the general approach for the PTP tests be? @jnunyez might be able to say.

10' seems too much. If we need to wait more than for instance thirty seconds  after config is applied I would say something is wrong even if after that time let's say 4' or 5' there are logs in the GPS. It is worth a try. Let's put a minute for instance and try.

Reply:
Ok I changed it to 1 minute and it looks good.